### PR TITLE
Apply Govspeak JS enhancements

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/VDgVNurM/544-extract-govspeak-javascript-functionality-to-govukpublishingcomponents

This is a breaking change for any apps that use govuk_publishing_components and have their own copy of enhancements. The only app I expect this to affect is static. For other applications there is a delicate time window where they'll possibly have some renderings broken by static when they upgrade to this gem, there doesn't seem to be a particularly nice way around that so I'll just plan to deploy lots of things at similar times.

This copies files from govuk_frontend_toolkit and static to enable the Govspeak JS enhancements to work without static running. 

Before:

screen shot 2019-02-15 at 12 43 40
screen shot 2019-02-15 at 12 43 25

After:

screen shot 2019-02-15 at 12 43 02
screen shot 2019-02-15 at 12 42 55

An intention of this PR is to change static so it uses files from here for applying it's own Govspeak enhancements (for pages that haven't used the govspeak component from here) so that the code can then be removed from static.

As the JS is rather legacy adding it in has been a little bit grim, I maintained the heavy use of jQuery and battled a bit for it to have tests that regularly pass.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,13 @@
   useful summary for people upgrading their application, not a replication
   of the commit log.
 
+## Unreleased
+
+* BREAKING: Apply JS enhancements to govspeak. Apps that already apply this to
```
Comment (andysellick) [inline CHANGELOG.md]: The changelog should just list the names of the PRs, not provide full detail.

File: app/assets/stylesheets/govuk_publishing_components/components/govspeak/_media-player.scss [hunk 2]
```diff
@@ -11,5 +11,12 @@
 
   .media-player {
     @include media-player;
+
+    // We need a specific styling on form elements otherwise it falls through
+    // to the browser default, this is needed in govuk_publishing_components
+    // as there is not a global reset like there is in govuk_frontend_toolkit
+    button {
+      @include core-16;
```
Comment (alex-ju) [inline app/assets/stylesheets/govuk_publishing_components/components/govspeak/_media-player.scss]: can we use the `govuk-frontend` mixins instead? `@include govuk-font($size: 16);`

File: app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss [hunk 3]
```diff
@@ -11,6 +11,8 @@
 .gem-c-govspeak {
   @include govuk-text-colour;
 
+  @include core-16;
```
Comment (alex-ju) [inline app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: `@include govuk-font($size: 16);`?

Comment (kevindew) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: I didn't do that then as I don't want to have sections/files where we have some govuk-frontend font mixins and some govuk_frontend_toolkit ones. It makes it look like they have special meaning rather than one legacy and one not.

Really I should have just stopped at that comment and said "no" but no I had to had try be a hero and I've converted all the govspeak font and media queries to use govuk-frontend mixins. I've put this in as a separate commit so it's easy to chop off this if it looks problematic.

I've put together a diff that shows how the resulting CSS changes if you want to have a scan of that and see if that introduces any areas of concern: https://www.diffchecker.com/1IHtaMg8

Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: Wow.. you really went down the 'hero' road. I'll do some testing, but looks great so far.

Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: All good, I think we can get this merged.

Comment (kevindew) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: Great 👍 - before we merge do you want to see if there's anything we can about the barchart selector here?

Comment (alex-ju) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss]: Seems to be working fine as it is. I'd like to have a look at the `magnaCharta` repo and update the script to prevent is from being initialised twice (which was a good find-out) - I made a note to look into it.

File: app/assets/stylesheets/govuk_publishing_components/components/govspeak/_fraction.scss [hunk 4]
```diff
@@ -10,7 +10,7 @@
   .fraction {
     sup,
     sub {
-      @include core-14;
+      @include govuk-font($size: 16);
```
Comment (36degrees) [inline app/assets/stylesheets/govuk_publishing_components/components/govspeak/_fraction.scss]: Is this change intentional?

Comment (kevindew) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_fraction.scss]: It is not! thanks 👍

Comment (kevindew) [reply app/assets/stylesheets/govuk_publishing_components/components/govspeak/_fraction.scss]: fixed in b49f80d7 - as well as another small issue

**Comment by kevindew** [comment]
@alex-ju in light of Olly's comment would you mind just doing a quick second review that all the govspeak CSS changes look right to you? then I'll hit the merge button 😬

**Comment by alex-ju** [comment]
@kevindew yep, will do. seems like my eyes are letting me down 👓 . Thanks for catching that @36degrees!

**Comment by kevindew** [comment]
Thanks @alex-ju 👍
